### PR TITLE
Simplify condition to deploy on main

### PR DESCRIPTION
**Pull request by AidanDelaney**
Use the syntax from the Github Actions documentation

https://docs.github.com/en/actions/learn-github-actions/expressions

### CODE REVIEW

File: .github/workflows/build-and-deploy.yml [hunk 1]
```diff
@@ -42,12 +42,12 @@ jobs:
       - name: Build
         run: PACK_CMD=./pack make build-linux build-wine
       - uses: azure/docker-login@v1
-        if: (github.event_name == 'repository_dispatch') || (github.event_name == 'push' && github.ref == 'refs/heads/main')
```
Comment (samj1912) [inline .github/workflows/build-and-deploy.yml:45]: Why remove the dispatch?

Comment (AidanDelaney) [reply .github/workflows/build-and-deploy.yml:45]: AFAIK this step should run only on the `main` branch.  The workflow is [executed on repository_dispatch](https://github.com/buildpacks/samples/blob/8202f569ee8823604f28809eb4c930528e9d4844/.github/workflows/build-and-deploy.yml#L10), but I believe the step should only run on the `main` branch.

**Comment by natalieparellano** [comment]
Do we know why CircleCI is failing with `ERROR: failed to build: failed to fetch builder image 'index.docker.io/cnbs/sample-builder:jammy': image 'index.docker.io/cnbs/sample-builder:jammy' does not exist on the daemon: not found`?

**Comment by AidanDelaney** [comment]
We have not sucessfully published a :jammy image yet, because the GH
workflow keeps skipping the deploy step. Hence, in this PR, we're
simplifying the condition.

**Review by samj1912 — APPROVED** [review]

**Pull request merged**